### ISSUE TITLE
A bug from svelte-spa-router and scroll

### ISSUE THREAD

**Issue by mowtwo** [ItalyPaleAle/svelte-spa-router#278]
When I scroll on a page and then switch routes, scroll on this new page, and then use the browser's native back button to return, I will find that the page content has not been properly destroyed

This is a [minimal case](https://github.com/mowtwo/svelte-spa-router-bug)

**Comment by ItalyPaleAle** [comment]
Hi @mowtwo I tried cloning your repo but I can't reproduce the issue. Going back to the previous page, I see the same content I saw at the beginning.

What is the issue you're experiencing?

**Comment by ItalyPaleAle** [comment]
Can you share a screenshot/screencap of the issue? This is what it looks like on my laptop:

https://user-images.githubusercontent.com/43508/193609602-048de8eb-f49a-4a83-b0da-665cd7ace750.mov

**Comment by mowtwo** [comment]
@ItalyPaleAle 
I'd like to share screenshot/screencap 😯, but I'm on national holiday,and I may need to wait five days to share screenshot/screencap on the company's computer.
This bug appears in one of my projects, it may also be related to the transition function, because after I delete some transition, this bug will not reappear.
If you want to reproduce on a MacBook with good performance, you can set the `length` of the `Array. from that creates the `list` in the `Sub.svelte``
 file to a larger value, such as 200 or 300

**Comment by mowtwo** [comment]
Hi @ItalyPaleAle ,This video is the process of reproducing bugs
1. Scroll once on the first page
2. Click the sub link
3. Scroll on the sub page once to make the scroll top not equal to the first page
4. Click the browser's return button




https://user-images.githubusercontent.com/41664636/194679729-9c2d0251-b361-4cbc-8e89-9b1f7e35178d.mp4

**Comment by mowtwo** [comment]
Hi @ItalyPaleAle  ,This video is the process of reproducing bugs

1. Scroll once on the first page
2. Click the sub link
3. Scroll once on the sub page to make the scroll top larger than the first page
4. Click the browser's return button


https://user-images.githubusercontent.com/41664636/194680124-4f39feaf-104b-4ced-9d4c-1eea7251112b.mp4

**Comment by ItalyPaleAle** [comment]
Thanks for sharing that. Interestingly, I can reproduce the issue too but I am not sure this is a bug in the router. I was able to confirm the router is detecting the change in the route and notifying that.

I am leaning towards this possibly being a bug in Svelte itself?

**Comment by da-z** [comment]
Indeed, thank you! I just had the same glitch (on route change both old and new inner contents were loaded for a very short while, equivalent to the svelte transition time I had set on the component. After removing transition I no longer see the glitch.

**Comment by mowtwo** [comment]
> Thanks for sharing that. Interestingly, I can reproduce the issue too but I am not sure this is a bug in the router. I was able to confirm the router is detecting the change in the route and notifying that.
> 
> I am leaning towards this possibly being a bug in Svelte itself?

I agree with you, but I can't find the specific cause of the problem correctly at present. I hope you can help me find the problem. Thank you.

**Comment by ItalyPaleAle** [comment]
@mowtwo may be related to transitions given @da-z’s feedback. Also it wouldn’t be the first time transitions in svelte have caused bugs like these.
